dashboard/test/controllers/sections_controller_test.rb

```diff
@@ -196,6 +200,38 @@ class SectionsControllerTest < ActionController::TestCase
     assert_equal(false, section_with_script.hidden)
   end
 
+  test "update: course_id is not updated if invalid" do
+    Course.stubs(:valid_course_id?).returns(false)
+
+    sign_in @teacher
+    section = create(:section, user: @teacher, course_id: nil)
+
+    post :update, params: {
+      id: section.id,
+      course_id: 1,
+    }
+    section.reload
+    assert_response :success
+    assert_nil section.course_id
+    refute_equal 1, section.course_id
```

Comment:
redundant check after asserting nil? here and below

Reply:
haha good point. i'll update 👍